google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcStorageOptions.java

```diff
@@ -94,10 +98,20 @@ Duration getTerminationAwaitDuration() {
 
   @InternalApi
   StorageSettings getStorageSettings() throws IOException {
-    URI uri = URI.create(getHost());
+    String endpoint = getHost();
+    URI uri = URI.create(endpoint);
     String scheme = uri.getScheme();
-    int port = uri.getPort() > 0 ? uri.getPort() : scheme.equals("http") ? 80 : 443;
-    String endpoint = String.format("%s:%d", uri.getHost(), port);
+    int port = uri.getPort();
+    // Gax routes the endpoint into a method which can't handle schemes, unless for direct path
+    // try and strip here if we can
+    switch (scheme) {
+      case "http":
+        endpoint = String.format("%s:%s", uri.getHost(), port > 0 ? port : 80);
+        break;
+      case "https":
+        endpoint = String.format("%s:%s", uri.getHost(), port > 0 ? port : 443);
```

Comment:
Implementer's choice, just looking at alternative syntax to avoid the implicit-fallthrough `default` case:
```java
private String resolveEndpoint(String originalEndpoint) {
  URI uri = URI.create(originalEndpoint);

  int defaultPort = scheme.equals("http") ? 80 : 443;
  int explicitPort = uri.getPort();
  int port = explicitPort > 0 ? explicitPort : defaultPort;

  // Gax routes the endpoint into a method which can't handle schemes.
  // Unless for direct path, try and strip here if we can.
  return (schema.equals("http") || schema.equals("https"))
    ? String.format("%s:%s", uri.getHost(), port)
    : originalEndpoint;
}

@InternalApi
StorageSettings getStorageSettings() throws IOException {
  String endpoint = resolveEndpoint(getHost());